### ISSUE TITLE
Problem with HSL solvers since updating to MacOS 13.0

### ISSUE THREAD

**Issue by this-josh** [jump-dev/Ipopt.jl#339]
I installed the HSL solvers while on MacOS 12.x and they worked fine, now on MacOS 13.0 I'm finding the following:
1. Launching Julia from the HSL library path (/Users/jak121/CoinOR/HSL/lib) gives me access to the HSL linear solvers
2. Using `set_optimizer_attribute(model, "hsllib", "/Users/jak121/CoinOR/HSL/lib/libhsl.dylib")` works
3. Running`export DL_LOAD_PATH=/Users/jak121/CoinOR/HSL/lib` and moving to a separate dir gives
```shell
Exception of type: DYNAMIC_LIBRARY_FAILURE in file "Common/IpLibraryLoader.cpp" at line 67:
 Exception message: dlopen(libhsl.dylib, 0x0002): tried: 'libhsl.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibhsl.dylib' (no such file), '/Users/jak121/.julia/artifacts/5f87ee5a0999062b66c1db490761d65343e0613f/lib/./libhsl.dylib' (no such file), '/Users/jak121/.julia/juliaup/julia-1.8.2+0.aarch64/lib/libhsl.dylib' (no such file), '/usr/lib/libhsl.dylib' (no such file, not in dyld cache), 'libhsl.dylib' (no such file)

EXIT: Library loading failure.
```
which suggests to me that `DL_LOAD_PATH` isn't the correct environment variable to set as it doesn't even seem to be trying this path. I have also tried setting `LD_LIBRARY_PATH` and this does not work. 

After step 3 if I run Julia I get
```Julia
julia> using Libdl

julia> dlopen("sys")
Ptr{Nothing} @0x00000000a1e14350

julia> dlopen("libhsl")
ERROR: could not load library "libhsl"
dlopen(libhsl.dylib, 0x0001): symbol not found in flat namespace '_metis_nodend_'
Stacktrace:
 [1] dlopen(s::String, flags::UInt32; throw_error::Bool)
   @ Base.Libc.Libdl ./libdl.jl:117
 [2] dlopen (repeats 2 times)
   @ ./libdl.jl:116 [inlined]
 [3] top-level scope
   @ REPL[3]:1

julia> using JuMP,Ipopt

julia> begin
           m = Model(optimizer_with_attributes(Ipopt.Optimizer, "linear_solver"=>"ma27"))
           @variable(m, 0<= x <= 1)
           @objective(m, Min, x)
           optimize!(m)
       end
Exception of type: DYNAMIC_LIBRARY_FAILURE in file "Common/IpLibraryLoader.cpp" at line 67:
 Exception message: dlopen(libhsl.dylib, 0x0002): tried: 'libhsl.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibhsl.dylib' (no such file), '/Users/jak121/.julia/artifacts/5f87ee5a0999062b66c1db490761d65343e0613f/lib/./libhsl.dylib' (no such file), '/Users/jak121/.julia/juliaup/julia-1.8.2+0.aarch64/lib/libhsl.dylib' (no such file), '/usr/lib/libhsl.dylib' (no such file, not in dyld cache), 'libhsl.dylib' (no such file)

EXIT: Library loading failure.

julia> dlopen("libhsl")
Ptr{Nothing} @0x00000000a1e15e10
```

So before using Ipopt `dlopen("libhsl")` fails, then I run Ipopt which fails to find `libhsl`, and after this `dlopen("libhsl")` works. 

Is there another environment variable I should now be setting?

This may be unrelated to the MacOS update, but I don't think I've changed anything else since yesterday (when I was on 12.x).

**Comment by odow** [comment]
> Using set_optimizer_attribute(model, "hsllib", "/Users/jak121/CoinOR/HSL/lib/libhsl.dylib") works

We should update the documentation in the README to this.

**Comment by this-josh** [comment]
I’ll open a PR  (given we’ve had an [issue](https://github.com/jump-dev/JuMP.jl/pull/3113) to discuss this 😊) tomorrow.

it would be nice if it didn’t have to be set every time but I’m not sure this is possible.

**Comment by odow** [comment]
Closed by #340
